sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -214,18 +263,57 @@
  */
 @Experimental
 public class KafkaIO {
+
+  /**
+   * Attempt to infer a {@link Coder} by extracting the type of the deserialized-class
+   * from the deserializer argument using the {@link Coder} registry.
+   */
+  public static <T> Coder<T> inferCoder(CoderRegistry coderRegistry,
```

Comment:
This shouldn't be public - better make it package-local (no access modifier) and mark `@VisibleForTesting`.